src/gui/viewer.cpp

```diff
@@ -390,6 +394,32 @@ void ViewerMainWindow::reload_file() {
 	document()->load_file(document()->path().c_str());
 }
 
+void ViewerMainWindow::import_file() {
+	auto filename = QFileDialog::getOpenFileName(this, tr("Open Image"), QString(), "JPG Files (*.jpg)").toUtf8();
+
+	//	ImageImportDialog dialog(this);
+	//	dialog.exec();
+
+	if (filename.isEmpty())
+		return;
+
+	auto image = imglib::image_begin(filename);
+
+	auto data = imglib::image_load(image);
+
+	auto file = new CVTFFile();
+
+	// TODO: Properly handle different image formats
+	file->Create(data.info.w, data.info.h, 1, 1, 1, VTFImageFormat::IMAGE_FORMAT_RGB888);
```

Comment:
`data` will not always be in RGB888, so this will not work if the file is HDR or has an alpha channel.

You can choose the correct format using an if statement like this:
```cpp
auto comps = data.info.comps;
auto w = data.info.w;
auto h = data.info.h;
VTFImageFormat format;
switch(data.info.type) {
case: imglib::UInt16:
    format = RGBA16161616F; break; // @TODO: Add RGBA16 support
case imglib::Float:
    format = (comps == 3) ? RGB323232F : RGBA32323232F; break;
default:
    format = (comps == 3) ? RGB888 : RGBA8888;
}
```

We should probably add a generic util function to `imglib` to get a compatible VTF image format for the image data.

The `TODO` can probably be removed after adding this too. The user can select the desired format from the format dropdown list in the viewer.

Reply:
Sure thing! I will make definitely make that generic function in a future.